### ISSUE TITLE
`term_replaced_by` short forms

### ISSUE THREAD

**Issue by jamesamcl** [EBISPOT/ols4#154]
In OLS3 the `term_replaced_by` field is populated from the `http://purl.obolibrary.org/obo/IAO_0100001` property.

The value seems to always be IRI. However, sometimes the IRI is specified correctly as a named RDF node:

```xml
    <owl:Class rdf:about="http://purl.obolibrary.org/obo/CHEBI_22584">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2002/07/owl#Thing"/>
        <obo1:IAO_0000231 rdf:resource="http://purl.obolibrary.org/obo/IAO_0000227"/>
        <obo1:IAO_0100001 rdf:resource="http://purl.obolibrary.org/obo/CHEBI_2762"/>
        <oboInOwl:hasDbXref>MeSH:D000968</oboInOwl:hasDbXref>
        <owl:deprecated rdf:datatype="http://www.w3.org/2001/XMLSchema#boolean">true</owl:deprecated>
    </owl:Class>
```

While sometimes it is specified as a string:

```xml
    <owl:Class rdf:about="http://www.orpha.net/ORDO/Orphanet_99961">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2002/07/owl#Thing"/>
        <obo1:IAO_0100001 rdf:datatype="http://www.w3.org/2001/XMLSchema#string">http://purl.obolibrary.org/obo/MONDO_0011559</obo1:IAO_0100001>
        <efo:obsoleted_in_version rdf:datatype="http://www.w3.org/2001/XMLSchema#string">3.44.0</efo:obsoleted_in_version>
        <efo:reason_for_obsolescence rdf:datatype="http://www.w3.org/2001/XMLSchema#string">Replaced with Mondo term. Use: http://purl.obolibrary.org/obo/MONDO_0011559</efo:reason_for_obsolescence>
        <rdfs:label rdf:datatype="http://www.w3.org/2001/XMLSchema#string">obsolete_benign recurrent intrahepatic cholestasis type 2</rdfs:label>
        <owl:deprecated rdf:datatype="http://www.w3.org/2001/XMLSchema#boolean">true</owl:deprecated>
    </owl:Class>
```

The OLS4 JSON format does not really differentiate between URIs and strings. This hasn't really been a problem so far. For example, in json2neo we decide whether something is an edge in Neo4j based simply on whether its value is the IRI of another node, without caring at all about the datatype of the value.

HOWEVER, in the OLS3 API, `term_replaced_by` is shortened to a short form iff the datatype is URI. If the datatype is string, it is not shortened. So the first returns (in OLS3) `"term_replaced_by": "CHEBI_2762"` while the second returns `"term_replaced_by": "http://purl.obolibrary.org/obo/MONDO_0011559"`. I can't reproduce this behaviour in OLS4 without differentiating between the URI and string datatypes.

In summary there are two issues here with OLS4:

1. The OLS4 data representation, which is supposed to be lossless, loses the datatype for URIs and therefore can't determine what is encoded as a URI and what is encoded as a string
2. This means we can't reproduce the OLS3 behaviour of shortening URIs but not strings

But also an issue with the ontologies encoding IRIs as strings instead of IRIs.

**Comment by jamesamcl** [comment]
More inconsistent randomness...

```xml
    <!-- http://www.orpha.net/ORDO/Orphanet_664 -->

    <owl:Class rdf:about="http://www.orpha.net/ORDO/Orphanet_664">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2002/07/owl#Thing"/>
        <obo1:IAO_0000115 rdf:datatype="http://www.w3.org/2001/XMLSchema#string">Ornithine carbamyl transferase deficiency is a disorder or urea cycle metabolism which clinical presentaton depends on the amount of residual enzyme activity, ranging from very severe neonatal hyperammonemic coma in males with no enzyme activity, to juvenile or adult hyperammonemic coma in males with residual enzyme activity. Females can be either asymptomatic or present with dislike for proteins to chronic vomiting, growth retardation, hypotonia, psychomotor retardation, hyperammonemic coma, or psychiatric disorders.</obo1:IAO_0000115>
        <obo1:IAO_0100001 rdf:datatype="http://www.w3.org/2001/XMLSchema#string">http://purl.obolibrary.org/obo/MONDO_0010703</obo1:IAO_0100001>
        <obo1:IAO_0100001>http://www.ebi.ac.uk/efo/EFO_0007409</obo1:IAO_0100001>
```

which one do we choose?!
